### PR TITLE
Reinstate a whole bunch of centaur tests. Let the shoes drop...!

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by geoffjentry** [comment]
@cjllanwarne Outside of someone saying "I like my shoes not fallen" is there any reason this needs a review?

File: backend/src/main/scala/cromwell/backend/io/GlobFunctions.scala [hunk 1]
```diff
@@ -8,16 +8,15 @@ import common.validation.ErrorOr.ErrorOr
...
 
 trait GlobFunctions extends IoFunctionSet {
 
   def callContext: CallContext
 
   def findGlobOutputs(call: TaskCallNode, jobDescriptor: BackendJobDescriptor): ErrorOr[List[WomGlobFile]] =
-    call.callable.outputs.flatTraverse[ErrorOr, WomGlobFile] {
-      _.expression.evaluateFiles(jobDescriptor.localInputs, this, WomAnyType) map {
+    call.callable.outputs.flatTraverse[ErrorOr, WomGlobFile] { outputDefinition =>
+      outputDefinition.expression.evaluateFiles(jobDescriptor.localInputs, this, outputDefinition.womType) map {
```
Comment (mcovarr) [inline backend/src/main/scala/cromwell/backend/io/GlobFunctions.scala]: what's up with the type change?

File: centaur/src/main/resources/standardTestCases/array_literal_locations.test [hunk 2]
```diff
@@ -14,6 +10,6 @@ metadata {
   workflowName: array_literal_locations
   status: Succeeded
   "outputs.array_literal_locations.array_literal_locations_i.out": "0:1:2"
-  "outputs.array_literal_locations.array_literal_locations_ii.0.out": "0"
-  "outputs.array_literal_locations.array_literal_locations_ii.1.out": "1"
+  "outputs.array_literal_locations.array_literal_locations_ii.out.0": "0"
+  "outputs.array_literal_locations.array_literal_locations_ii.out.1": "1"
```
Comment (mcovarr) [inline centaur/src/main/resources/standardTestCases/array_literal_locations.test]: 😄

File: backend/src/main/scala/cromwell/backend/io/GlobFunctions.scala (hunk 1 above)
Comment (cjllanwarne) [reply backend/src/main/scala/cromwell/backend/io/GlobFunctions.scala]: The FileEvaluator needs to know what it's coercing inputs into. Previously it was failing to read pair inputs from JSON (even though they didn't have any globs, they still need to go through the "does it have globs" method)

**Pull request merged**